components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/SyncManager.kt

```diff
@@ -59,12 +61,20 @@ object GlobalSyncableStoreProvider {
  */
 interface SyncDispatcher : Closeable, Observable<SyncStatusObserver> {
     fun isSyncActive(): Boolean
-    fun syncNow(startup: Boolean = false, debounce: Boolean = false)
+    fun syncNow(reason: SyncReason, debounce: Boolean = false)
     fun startPeriodicSync(unit: TimeUnit, period: Long)
     fun stopPeriodicSync()
     fun workersStateChanged(isRunning: Boolean)
 }
 
+enum class SyncReason {
+    STARTUP,
+    USER,
+    ENGINE_CHANGE,
+    // TODO this is really intended for internal use...
+    SCHEDULED
```

Comment:
Could use `sealed class` instead, and mark this as an `internal` object.

Reply:
What would be the benefit of using sealed classes?

Reply:
So that we may hide certain subtypes, such as `Scheduled`. They're useful internally, but not at the public API level.